include/devicetree/pwms.h

```diff
@@ -101,6 +101,79 @@ extern "C" {
  */
 #define DT_PWMS_LABEL(node_id) DT_PWMS_LABEL_BY_IDX(node_id, 0)
 
+/**
+ * @brief Get a pwm-controller node identifier from a pwms property at an index
+ *
+ * Example devicetree fragment:
+ *
+ *     pwm1: pwm-controller@... {
+ *             label = "PWM_1";
+ *     };
+ *
+ *     pwm2: pwm-controller@... {
+ *             label = "PWM_2";
+ *     };
+ *
+ *     n: node {
+ *             pwms = <&pwm1 1 PWM_POLARITY_NORMAL>,
+ *                    <&pwm2 3 PWM_POLARITY_INVERTED>;
+ *     };
+ *
+ * Example usage:
+ *
+ *     DT_PWMS_CTRL_BY_IDX(DT_NODELABEL(n), 0) // "PWM_1"
+ *     DT_PWMS_CTRL_BY_IDX(DT_NODELABEL(n), 1) // "PWM_2"
+ *
+ * @param node_id node identifier for a node with a pwms property
+ * @param idx logical index into pwms property
+ * @return  The node identifier for the node referenced at index "idx"
+ * @see DT_PROP_BY_PHANDLE_IDX()
+ */
+#define DT_PWMS_CTRL_BY_IDX(node_id, idx) \
```

Comment:
`CTLR`, not `CTRL`.

This is short for "controller" and is consistent with things like `DT_SPI_DEV_CS_GPIOS_CTLR` and `DT_SPI_DEV_CS_GPIOS_CTLR`.

The required devicetree.h and drivers/pwm.h changes are:

- add devicetree helper macros for getting the node identifier for the controller to this file (devicetree/pwms.h)
- add additional helper macros to drivers/pwm.h for getting the devices themselves, which use the new devicetree helper macros

For inspiration, please see similar changes for SPI in https://github.com/zephyrproject-rtos/zephyr/pull/30536/.